### ISSUE TITLE
CustomMap

### ISSUE THREAD

**Issue by GCHQDev404** [gchq/Gaffer#2024]
A Map that can serialise complicated Value and Key types, without the need for a custom implementation of map. Otherwise errors occur during serialisation with a regular map.

**Comment by GCHQDev404** [comment]
blocked by gh-2027

**Comment by GCHQDev404** [comment]
Not blocked now.

**Comment by d47853** [comment]
Merged into develop
